src/vm/threads.inl

```diff
@@ -36,7 +36,7 @@ EXTERN_C inline Thread* STDCALL GetThread()
 
 EXTERN_C inline AppDomain* STDCALL GetAppDomain()
 {
-    return gCurrentThreadInfo.m_pAppDomain;
+    return SystemDomain::System()->DefaultDomain();
```

Comment:
This does through multiple indirections. I guess it is still faster than TLS access, but it would be nice to have them.

Reply:
There is just one indirection: it translates to m_pSystemDomain->m_pDefaultDomain. But you are right, if we move the m_pDefaultDomain out of the SystemDomain, we can get a direct access. Let me do that.